src/confd/src/ietf-system.c

```diff
@@ -328,13 +331,22 @@ static int change_clock(sr_session_ctx_t *session, uint32_t sub_id, const char *
 
 	default:
 		return SR_ERR_OK;
-	}
 
-	/* XXX: add support also for /ietf-system:system/clock/timezone-utc-offset (deviation) */
+	}
+	tz_utc_offset = srx_get_str(session, XPATH_BASE_"/clock/timezone-utc-offset");
 	timezone = srx_get_str(session, XPATH_BASE_"/clock/timezone-name");
-	if (!timezone) {
-		ERROR("Failed reading timezone-name");
-		return SR_ERR_VALIDATION_FAILED;
+	if (!timezone && !tz_utc_offset) {
+		snprintf(tz_name,sizeof(tz_name),"Etc/UTC");
+		timezone = tz_name;
+	}
+
+	if(tz_utc_offset) {
+		int8_t offset = atol(tz_utc_offset);
+
+		/* When using Etc/GMT offsets, the +/- is inverted in tzdata. */
+		snprintf(tz_name,sizeof(tz_name), "Etc/GMT%s%.2d", offset>-1?"":"+", -offset);
+
+		timezone=tz_name;
```

Comment:
Minor, Linux coding style. E.g., missing space between if and (), missing spaces in assignment, and also no need for all the empty lines.